src/main/java/net/revelc/code/formatter/xml/lib/XmlDocumentFormatter.java

```diff
@@ -114,7 +116,11 @@ private void validateWellFormedness(String documentText) {
             reader.setErrorHandler(errorHandler);
             reader.parse(new InputSource(new StringReader(documentText)));
         } catch (Exception exception) {
-            throw new IllegalArgumentException(exception);
+            if (prefs.getSaxValidation().equals(FormattingPreferences.WARN)) {
+                System.out.println(exception.getMessage());
```

Comment:
Are there other places where the `System.out` is called directly? I wouldn't recommend calling `System.out` directly, over the use of a system logger, or at the very least, a global PrintStream object that is set to System.out at startup, so it can be managed in one place.

Reply:
I didn't see a logger, and there was a previous usage of using `System.out`, so that's what I went with: https://github.com/revelc/xml-formatter/blob/bd9e761e94ba7b02a6fb0d22f8ff7e6008ccb10e/src/main/java/net/revelc/code/formatter/xml/lib/XmlDocumentFormatter.java#L102

Also happy to revisit this choice, but it might be a separate issue?

Reply:
At the very least, it should use System.err for the warning so the caller can filter out STDERR.